sessions.md

```diff
@@ -0,0 +1,210 @@
+<a name="#top">Back to Top</a>
+
+# Sessions
+
+
+## Table of Contents
+
+* [Options](#Options)
+  * [httpOnly](#http-only)
+  * [secure](#secure)
+  * [path](#path)
+  * [domain](#domain)
+  * [validationStrategy](#validation-strategy)
+* [Incoming Request Handling](#incoming-request-handling)
+* [/me Handling](#me-handling)
+
+
+## Feature Description
+
+This document describes the how our framework integrations should handle user
+sessions and session management.
+
+There are a few things we'll do in respect to user sessions:
+
+- We'll always store an access and refresh token in the browser,
+these tokens will be obtained by using the password grant flow on
+the configured Stormpath application
+
+- This cookie will always be set with the `HttpOnly` flag set, meaning that no
+  Javascript access will be allowed to this cookie.
+
+- This cookie will be set with the `secure` flag if the application is in
+  production -- this ensures that no cookies will be set over plain old HTTP
+  (*leaking information to potential attackers*).
+
+- The cookies will always be named `accessToken` and `refreshToken`.
+
+- The Expiration time each cookie should be determined by the `exp` claim of
+the token, and this value is controlled by the Oauth Policy of the Application
+
+**NOTE**: We do NOT need to support server-side session storage, because the way
+this works is like so: a JWT is created which contains an account href.  The
+account href is persisted in our internal Stormpath SDK cache.  This means no
+server side session storage is necessary as the caching optimizations are
+already handled internally (*by us*).
+
+
+## <a name="Options"></a> Options
+
+This table is a list of all the options that are required by this feature.
+Detailed descriptions follow.  How the option names are translated into the
+framework language (e.g. to camel case, or not)? Is not specified here.
+
+```json
+{
+  "session": {
+    "httpOnly": true,
+    "secure": null,
+    "path": "/",
+    "domain": null,
+    "validationStrategy": "stormpath"
+  }
+}
+```
+
+#### <a name="http-only"></a> httpOnly
+
+This option determines whether or not this cookie can be read by Javascript code
+on the browser.
+
+**NOTE**: Disabling this option is a potential security issue, as your access
+tokens could be hijacked via XSS.
+
+<a href="#top">Back to Top</a>
+
+
+#### <a name="secure"></a> secure
+
+This option determines whether or not our cookies can be set over a non-https
+connection.  By default, this option is null, which means we will attempt to
```

Comment:
Instead of `null`, shouldn't they just not configure the option?  Null = nondeterministic value and can be problematic.  Makes me nervous.  Thoughts?